src/gc.c

```diff
@@ -2644,6 +2675,8 @@ mark: {
             else if (foreign_alloc)
                 objprofile_count(vt, bits == GC_OLD_MARKED, sizeof(jl_task_t));
             jl_task_t *ta = (jl_task_t*)new_obj;
+            latest_task = ta; // TODO: correct??
+            gc_heap_snapshot_record_root(ta, "task");
```

Comment:
```suggestion
```

Reply:
We need to add an edge from `task` to `stkbuf` (e.g. where we call `gc_setmark_buf_` I think.). To attribute the size of the stack to the task.

And then we want to add an edge at `gc_mark_stack_push`, where we start scanning the stack.

```
                nroots = gc_read_stack(&s->nroots, offset, lb, ub);
                assert(nroots <= UINT32_MAX);
                gc_mark_stackframe_t stackdata = {s, 0, (uint32_t)nroots, offset, lb, ub};
                gc_mark_stack_push(&ptls->gc_cache, &sp, gc_mark_laddr(stack),
                                   &stackdata, sizeof(stackdata), 1);
```

As I mentioned in slack. Not all tasks can be considered toplevel roots. Only the ones that are `current_task` in the `ptls` of each thread are roots, the others are "just" reachable objects, but each task then roots all the objects on its current stack. and it's exception stack,

Reply:
And you will want to instrument `jl_gc_queue_thread_local`